hazelcast-client/src/test/java/com/hazelcast/client/ClientConnectionListenerTest.java

```diff
@@ -0,0 +1,46 @@
+package com.hazelcast.client;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.ListenerConfig;
+import com.hazelcast.core.Client;
+import com.hazelcast.core.ClientListener;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.CountDownLatch;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
+public class ClientConnectionListenerTest extends HazelcastTestSupport {
```

Comment:
There was a `ClientListenerTest` already which is testing map, set, list etc listeners.
I've added 'connection' just because events are `clientConnected` and `clientDisconnected` :)